app/controllers/withdraw_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class WithdrawController < ApplicationController
+  def create
+    @document = Document.with_current_edition.find_by_param(params[:id])
+    edition = @document.current_edition
+    @public_explanation =
+      edition.withdrawn? ? edition.status&.details&.public_explanation : nil
+
+    if !current_user.has_permission?(User::PRE_RELEASE_FEATURES_PERMISSION)
+      render :withdraw
+      return
+    end
+
+    if current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION)
+      render :create
+    else
+      render :non_managing_editor, status: :forbidden
+    end
+  end
+
+  def new
+    unless user_has_permissions?
+      # FIXME: this shouldn't be an exception but we've not worked out the
+      # right response - maybe bad request or a redirect with flash?
+      raise "Can't withdraw an edition without permissions"
+    end
+
+    Document.transaction do
+      @document = Document.with_current_edition.lock!.find_by_param(params[:id])
+      public_explanation = params[:public_explanation]
+      issues = Requirements::WithdrawalChecker.new(public_explanation).pre_withdrawal_issues
+
+      if issues.any?
+        flash["alert_with_items"] = {
+          "title" => I18n.t!("withdraw.create.flashes.requirements"),
+          "items" => issues.items,
+        }
+
+        render :create
+      else
+        begin
+          raise "Cannot withdraw an edition that is not live" unless withdrawable?(@document)
```

Comment:
I think as this is essentially same check we do in UnpublishService with same result in UnpublishService this is superfluous.

I think my previous comments are the reason this ended up here, so firstly, Sorry! and secondly,  it'd make sense for us to do something better - which we should eventually do.

Do we want something like this on the `create` method? maybe even just a comment that says `# FIXME should check document is withdrawable`?

Reply:
Sorry, not quite sure what I need to do here - just leave a comment and not raise?

Reply:
> just leave a comment and not raise?

Yup 👍